### ISSUE TITLE
Add numeric/money refund column to items report

### ISSUE THREAD

**Issue by othergun** [philipmulcahy/azad#194]
Hi, the column "refund" stay at 0. Can you fix it ? Thanks for your work ! 🙏🔥

**Comment by fbietlot** [comment]
Hi @othergun it will be working if you use amazon.de in English language

@philipmulcahy there is actually the same error for amazon.fr (with no work around of switching to English langage)
I have traced and solved this on my forked version : https://github.com/fbietlot/azad/issues/5
I have other enhancement and/or bug fix, I would be happy to collaborate on this, even if It's the first time I'm using github

**Comment by philipmulcahy** [comment]
You’re gonna have to give me more to go on than that - remember that I don’t have access to your account to play with your orders. 
(Be thankful I don’t)
Instead, please read the bug reporting section in the README.

Thanks,

Philip

**Comment by psommerfeld** [comment]
I just found out about the extension! I don't see a refund column at all. Is there a way for this extension to record refunds?
<img width="1440" alt="Screenshot 2023-01-01 at 5 59 40 PM" src="https://user-images.githubusercontent.com/1151630/210186902-aa908dd4-22b5-4de2-a17e-1930e871ec30.png">

**Comment by philipmulcahy** [comment]
Looking at your screenshot, I think you have selected the “show items not orders” view. The other (row-per-order) has a refund column.I hope this is what you need.Yours,PhilipOn 1 Jan 2023, at 23:00, psommerfeld ***@***.***> wrote:﻿
I just found out about the extension! I don't see a refund column at all. Is there a way for this extension to record refunds?

**Comment by psommerfeld** [comment]
You're right! Yes that's what I need! I thought "show items not orders"
would be better but it looks like this view has all the info I need, plus
more!
Peter

**Comment by richardbowman** [comment]
Just read this thread, now that Amazon killed the Order History Report (again), I think a lot of us will be back to try to make your excellent extension work. Amazon on the order details screen shows the specific items that have been refunded. Is it possible to add refund amount to the items report? Would be immensely useful.

**Comment by philipmulcahy** [comment]
It’s been a while since I looked at that bit of code, but I fear that your request is very tough to honour.

**Comment by polychromeuganda** [comment]
There are several refunds on the order page but there is distinct structure
The order detail page has a total refund with a popover breaking the total down merchandise, shipping and tax refunded,
<div class="a-popover-preload" id="a-popover-orderRefundBreakdown">
    <div class="a-row">
        <div class="a-column a-span9">
            <span class="a-color-success">
                Item(s) refund:
            </span>
        </div>
        <div class="a-column a-span3 a-text-right a-span-last">
            <span class="a-color-success">
                $115.26
            </span>
        </div>
    </div>
    <div class="a-row">
        <div class="a-column a-span9">
            <span class="a-color-success">
                Tax refund:
            </span>
        </div>
        <div class="a-column a-span3 a-text-right a-span-last">
            <span class="a-color-success">
                $7.20
            </span>
        </div>
    </div>
    <div class="a-row a-spacing-top-mini">
        <div class="a-column a-span9">
            <span class="a-color-success a-text-bold">
                Refund Total:
            </span>
        </div>
        <div class="a-column a-span3 a-text-right a-span-last">
            <span class="a-color-success a-text-bold">
                $122.46
            </span>
        </div>
    </div>

The shipments have refunds with a clear pattern
within:      <div class="a-box shipment">
After:        <a href="javascript:void(0)" class="a-popover-trigger a-declarative"><span class="a-color-success a-text-bold">Refund for this return</span><i class="a-icon a-icon-popover"></i></a>

[NB:  do not use "Refund for this return" alone or find the 2nd instance, "Refund for this return/n" precedes it]

This will be present:
       <div class="a-column a-span3 a-text-right a-span-last">
            <span class="a-color-success a-text-bold">
                $80.04
            </span>
        </div>

The phrase "refund for this return" appears to be used for each item no matter how many were part of the same return shipment. In recent years the CSV download reports refunds and returns have had a 1:1 mapping because both are always single item records.

**Comment by polychromeuganda** [comment]
The syntax doesn't appear to be all that daunting per order (which is already done) or per item, 
Amazon has regarded each returned item as a separate return for at least 8 years. 
At the order level the refund transactions correspond 1:1 with refunds at the item level and don't need to be dealt with.
At the item level
The sequence "language specific phrase" < don't care text > "local currency symbol" amount
is repeated twice in short succession creating its own check.

For the USA "language specific phrase" is "Refund for this return"
            and "local currency symbol" is "$"

"Refund for this return:\n" precedes  "\n                $80.04\n" 
and then its repeated 
< a href="javascript:void(0)" class="a-popover-trigger a-declarative" >< span class="a-color-success a-text-bold" >Refund for this return< /span >< i class="a-icon a-icon-popover" >< /i >< /a >
< /span >
           < /span >
        < /div >
        < div class="a-column a-span3 a-text-right a-span-last" >
            < span class="a-color-success a-text-bold" >
                $80.04
            < /span >
        < /div >
    < /div >
    < /div >

**Comment by philipmulcahy** [comment]
there seems to be at least one more refund related bug around - would anyone consider sending me an order debug dump for an order that's missing the refund in the output table, together with a marked up screenshot of what it _should_ look like.
Send it to the extension's email address, not here (but put the issue number (194) in the subject. Not promising to move fast, but fixing existing stuff seems more important than shiny shiny (which is what I've been focussing on recently).

**Comment by philipmulcahy** [comment]
I'm going to retitle this to reflect what I believe to be the remaining gap after the discussion above (and change from bug to enhancement).

**Comment by philipmulcahy** [comment]
and then I re-found #178..., which seems to be a prior art duplicate... this one gets closed and further discussion should happen there.
